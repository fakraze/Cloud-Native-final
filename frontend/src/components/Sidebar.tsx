import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { 
  Store, 
  ShoppingBag, 
  History, 
  User, 
  BarChart3,
  UtensilsCrossed,
  CreditCard
} from 'lucide-react';
import { useAuthStore } from '../store/authStore';

const Sidebar: React.FC = () => {
  const { user } = useAuthStore();
  const location = useLocation();

  const isAdmin = user?.role === 'admin';

  const employeeNavItems = [
    { name: 'Restaurants', path: '/restaurant', icon: Store },
    { name: 'Ongoing Orders', path: '/order', icon: ShoppingBag },
    { name: 'Order History', path: '/order-history', icon: History },
    { name: 'Personal', path: '/personal', icon: User },
  ];

  const adminNavItems = [
    { name: 'Dashboard', path: '/admin', icon: BarChart3 },
    { name: 'Restaurant Management', path: '/admin/restaurants', icon: UtensilsCrossed },
    { name: 'Order Management', path: '/admin/orders', icon: ShoppingBag },
    { name: 'POS Interface', path: '/admin/pos', icon: CreditCard },
  ];

  const navItems = isAdmin ? adminNavItems : employeeNavItems;

  return (
    <aside className="bg-white w-64 min-h-screen shadow-sm border-r border-gray-200">
      <div className="p-6">
        <div className="flex items-center space-x-3 mb-8">
          <div className="bg-blue-600 rounded-lg p-2">
            <UtensilsCrossed className="h-6 w-6 text-white" />
          </div>
          <div>
            <h2 className="text-lg font-semibold text-gray-900">
              {isAdmin ? 'Admin Panel' : 'Order System'}
            </h2>
            <p className="text-sm text-gray-500">{user?.name}</p>
          </div>
        </div>

        <nav className="space-y-2">
          {navItems.map((item) => {
            const isActive = location.pathname === item.path || 
              (item.path !== '/admin' && location.pathname.startsWith(item.path));

            return (
              <Link
                key={item.path}
                to={item.path}
                className={`flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors ${
                  isActive
                    ? 'bg-blue-50 text-blue-700 border-r-2 border-blue-700'
                    : 'text-gray-700 hover:bg-gray-50'
                }`}
              >
                <item.icon className="h-5 w-5" />
                <span className="font-medium">{item.name}</span>
              </Link>
            );
          })}
        </nav>

        {!isAdmin && (
          <div className="mt-8 pt-6 border-t border-gray-200">
            <div className="text-sm text-gray-500 mb-2">Quick Actions</div>
            <Link
              to="/cart"
              className="flex items-center space-x-3 px-3 py-2 text-gray-700 hover:bg-gray-50 rounded-lg transition-colors"
            >
              <ShoppingBag className="h-5 w-5" />
              <span className="font-medium">View Cart</span>
            </Link>
          </div>
        )}
      </div>
    </aside>
  );
};

export { Sidebar };
