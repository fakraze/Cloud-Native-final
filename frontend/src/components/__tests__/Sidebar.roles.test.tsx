import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render } from '../../test/test-utils';
import { Sidebar } from '../Sidebar';

const mockAuthStore = vi.hoisted(() => ({
  useAuthStore: vi.fn()
}));

vi.mock('../../store/authStore', () => mockAuthStore);

describe('Sidebar Component - Different Roles', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render admin sidebar navigation', () => {
    mockAuthStore.useAuthStore.mockReturnValue({
      user: {
        id: '1',
        username: 'admin',
        role: 'admin'
      }
    });

    const { container } = render(<Sidebar />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render staff sidebar navigation', () => {
    mockAuthStore.useAuthStore.mockReturnValue({
      user: {
        id: '1',
        username: 'staff',
        role: 'staff'
      }
    });

    const { container } = render(<Sidebar />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render customer sidebar navigation', () => {
    mockAuthStore.useAuthStore.mockReturnValue({
      user: {
        id: '1',
        username: 'customer',
        role: 'customer'
      }
    });

    const { container } = render(<Sidebar />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
