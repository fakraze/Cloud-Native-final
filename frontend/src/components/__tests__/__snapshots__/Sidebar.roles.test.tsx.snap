// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar Component - Different Roles > should render admin sidebar navigation 1`] = `
<aside
  class="bg-white w-64 min-h-screen shadow-sm border-r border-gray-200"
>
  <div
    class="p-6"
  >
    <div
      class="flex items-center space-x-3 mb-8"
    >
      <div
        class="bg-blue-600 rounded-lg p-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-utensils-crossed h-6 w-6 text-white"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m16 2-2.3 2.3a3 3 0 0 0 0 4.2l1.8 1.8a3 3 0 0 0 4.2 0L22 8"
          />
          <path
            d="M15 15 3.3 3.3a4.2 4.2 0 0 0 0 6l7.3 7.3c.7.7 2 .7 2.8 0L15 15Zm0 0 7 7"
          />
          <path
            d="m2.1 21.8 6.4-6.3"
          />
          <path
            d="m19 5-7 7"
          />
        </svg>
      </div>
      <div>
                    
        <h2
          class="text-lg font-semibold text-gray-900"
        >
          Admin Panel
        </h2>
        <p
          class="text-sm text-gray-500"
        />
      </div>
    </div>
    <nav
      class="space-y-2"
    >
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/admin"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-chart-column h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 3v16a2 2 0 0 0 2 2h16"
          />
          <path
            d="M18 17V9"
          />
          <path
            d="M13 17V5"
          />
          <path
            d="M8 17v-3"
          />
        </svg>
        <span
          class="font-medium"
        >
          Dashboard
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/admin/restaurants"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-utensils-crossed h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m16 2-2.3 2.3a3 3 0 0 0 0 4.2l1.8 1.8a3 3 0 0 0 4.2 0L22 8"
          />
          <path
            d="M15 15 3.3 3.3a4.2 4.2 0 0 0 0 6l7.3 7.3c.7.7 2 .7 2.8 0L15 15Zm0 0 7 7"
          />
          <path
            d="m2.1 21.8 6.4-6.3"
          />
          <path
            d="m19 5-7 7"
          />
        </svg>
        <span
          class="font-medium"
        >
          Restaurant Management
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/admin/orders"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-shopping-bag h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
          />
          <path
            d="M3 6h18"
          />
          <path
            d="M16 10a4 4 0 0 1-8 0"
          />
        </svg>
        <span
          class="font-medium"
        >
          Order Management
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/admin/send-notification"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-bell h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.268 21a2 2 0 0 0 3.464 0"
          />
          <path
            d="M3.262 15.326A1 1 0 0 0 4 17h16a1 1 0 0 0 .74-1.673C19.41 13.956 18 12.499 18 8A6 6 0 0 0 6 8c0 4.499-1.411 5.956-2.738 7.326"
          />
        </svg>
        <span
          class="font-medium"
        >
          Send Notification
        </span>
      </a>
    </nav>
            
  </div>
</aside>
`;

exports[`Sidebar Component - Different Roles > should render customer sidebar navigation 1`] = `
<aside
  class="bg-white w-64 min-h-screen shadow-sm border-r border-gray-200"
>
  <div
    class="p-6"
  >
    <div
      class="flex items-center space-x-3 mb-8"
    >
      <div
        class="bg-blue-600 rounded-lg p-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-utensils-crossed h-6 w-6 text-white"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m16 2-2.3 2.3a3 3 0 0 0 0 4.2l1.8 1.8a3 3 0 0 0 4.2 0L22 8"
          />
          <path
            d="M15 15 3.3 3.3a4.2 4.2 0 0 0 0 6l7.3 7.3c.7.7 2 .7 2.8 0L15 15Zm0 0 7 7"
          />
          <path
            d="m2.1 21.8 6.4-6.3"
          />
          <path
            d="m19 5-7 7"
          />
        </svg>
      </div>
      <div>
                    
        <h2
          class="text-lg font-semibold text-gray-900"
        >
          Order System
        </h2>
        <p
          class="text-sm text-gray-500"
        />
      </div>
    </div>
    <nav
      class="space-y-2"
    >
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/restaurant"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-store h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2 7 4.41-4.41A2 2 0 0 1 7.83 2h8.34a2 2 0 0 1 1.42.59L22 7"
          />
          <path
            d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
          />
          <path
            d="M15 22v-4a2 2 0 0 0-2-2h-2a2 2 0 0 0-2 2v4"
          />
          <path
            d="M2 7h20"
          />
          <path
            d="M22 7v3a2 2 0 0 1-2 2a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 16 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 12 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 8 12a2.7 2.7 0 0 1-1.59-.63.7.7 0 0 0-.82 0A2.7 2.7 0 0 1 4 12a2 2 0 0 1-2-2V7"
          />
        </svg>
        <span
          class="font-medium"
        >
          Restaurants
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/order"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-shopping-bag h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
          />
          <path
            d="M3 6h18"
          />
          <path
            d="M16 10a4 4 0 0 1-8 0"
          />
        </svg>
        <span
          class="font-medium"
        >
          Ongoing Orders
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/order-history"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-history h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 12a9 9 0 1 0 9-9 9.75 9.75 0 0 0-6.74 2.74L3 8"
          />
          <path
            d="M3 3v5h5"
          />
          <path
            d="M12 7v5l4 2"
          />
        </svg>
        <span
          class="font-medium"
        >
          Order History
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/payment"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-credit-card h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="14"
            rx="2"
            width="20"
            x="2"
            y="5"
          />
          <line
            x1="2"
            x2="22"
            y1="10"
            y2="10"
          />
        </svg>
        <span
          class="font-medium"
        >
          Payment
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/personal"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-user h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
          />
          <circle
            cx="12"
            cy="7"
            r="4"
          />
        </svg>
        <span
          class="font-medium"
        >
          Personal
        </span>
      </a>
    </nav>
            
    <div
      class="mt-8 pt-6 border-t border-gray-200"
    >
      <div
        class="text-sm text-gray-500 mb-2"
      >
        Quick Actions
      </div>
      <a
        class="flex items-center space-x-3 px-3 py-2 text-gray-700 hover:bg-gray-50 rounded-lg transition-colors"
        data-discover="true"
        href="/cart"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-shopping-bag h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
          />
          <path
            d="M3 6h18"
          />
          <path
            d="M16 10a4 4 0 0 1-8 0"
          />
        </svg>
        <span
          class="font-medium"
        >
          View Cart
        </span>
      </a>
    </div>
  </div>
</aside>
`;

exports[`Sidebar Component - Different Roles > should render staff sidebar navigation 1`] = `
<aside
  class="bg-white w-64 min-h-screen shadow-sm border-r border-gray-200"
>
  <div
    class="p-6"
  >
    <div
      class="flex items-center space-x-3 mb-8"
    >
      <div
        class="bg-blue-600 rounded-lg p-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-utensils-crossed h-6 w-6 text-white"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m16 2-2.3 2.3a3 3 0 0 0 0 4.2l1.8 1.8a3 3 0 0 0 4.2 0L22 8"
          />
          <path
            d="M15 15 3.3 3.3a4.2 4.2 0 0 0 0 6l7.3 7.3c.7.7 2 .7 2.8 0L15 15Zm0 0 7 7"
          />
          <path
            d="m2.1 21.8 6.4-6.3"
          />
          <path
            d="m19 5-7 7"
          />
        </svg>
      </div>
      <div>
                    
        <h2
          class="text-lg font-semibold text-gray-900"
        >
          Staff Panel
        </h2>
        <p
          class="text-sm text-gray-500"
        />
      </div>
    </div>
    <nav
      class="space-y-2"
    >
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/staff"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-chart-column h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 3v16a2 2 0 0 0 2 2h16"
          />
          <path
            d="M18 17V9"
          />
          <path
            d="M13 17V5"
          />
          <path
            d="M8 17v-3"
          />
        </svg>
        <span
          class="font-medium"
        >
          Dashboard
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/staff/orders"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-shopping-bag h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 2 3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4Z"
          />
          <path
            d="M3 6h18"
          />
          <path
            d="M16 10a4 4 0 0 1-8 0"
          />
        </svg>
        <span
          class="font-medium"
        >
          Order Management
        </span>
      </a>
      <a
        class="flex items-center space-x-3 px-3 py-2 rounded-lg transition-colors text-gray-700 hover:bg-gray-50"
        data-discover="true"
        href="/staff/pos"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-credit-card h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="14"
            rx="2"
            width="20"
            x="2"
            y="5"
          />
          <line
            x1="2"
            x2="22"
            y1="10"
            y2="10"
          />
        </svg>
        <span
          class="font-medium"
        >
          POS Interface
        </span>
      </a>
    </nav>
            
  </div>
</aside>
`;
